src/Compilers/CSharp/Test/Semantic/Semantics/OperatorTests.cs

```diff
@@ -14,6 +14,42 @@ namespace Microsoft.CodeAnalysis.CSharp.UnitTests
 {
     public partial class SyntaxBinderTests : CompilingTestBase
     {
+        [Fact, WorkItem(5419, "https://github.com/dotnet/roslyn/issues/5419")]
+        public void EnumBinaryOps()
+        {
+            string source = @"
+    [Flags]
+    internal enum TestEnum
+    {
+        None,
+        Tags,
+        FilePath,
+        Capabilities,
+        Visibility,
+        AllProperties = FilePath | Visibility
+    }
+    class C {
+         public void Foo(){
+            var x = TestEnum.FilePath | TestEnum.Visibility;
+        }
+    }
+";
+            var compilation = CreateCompilationWithMscorlib(source);
+            var tree = compilation.SyntaxTrees.Single();
+            var semanticModel = compilation.GetSemanticModel(tree);
+
+            var orNodes = tree.GetRoot().DescendantNodes().OfType<BinaryExpressionSyntax>().ToArray();
+            Assert.Equal(orNodes.Length, 2);
+
+            var insideEnumDefinition = semanticModel.GetSymbolInfo(orNodes[0]);
+            var insideMethodBody = semanticModel.GetSymbolInfo(orNodes[1]);
+
+            Assert.False(insideEnumDefinition.IsEmpty);
+            Assert.False(insideMethodBody.IsEmpty);
+
+            Assert.NotEqual(insideEnumDefinition, insideMethodBody);
```

Comment:
>Assert.NotEqual(insideEnumDefinition, insideMethodBody); [](start = 12, length = 56)

Please also assert what the symbols are, we usually do this by comparing result of ToTestDisplayString.